files/lib/chef_compat.rb

```diff
@@ -1,7 +1,22 @@
-require 'chef_compat/version'
-require 'chef_compat/resource'
-require 'chef_compat/property'
-require 'chef_compat/mixin/properties'
 
-module ChefCompat
+if Gem::Requirement.new(">= 12.0").satisfied_by?(Gem::Version.new(Chef::VERSION))
+
+  require 'chef_compat/version'
+  require 'chef_compat/resource'
+  require 'chef_compat/property'
+  require 'chef_compat/mixin/properties'
+
+  module ChefCompat
+  end
+
+else
+
+  class Chef
+    class Resource
```

Comment:
You may want to add property and resource_name as well, since those commonly show up early in resource declarations.